met-web/src/components/engagement/new/view/EngagementContentTabs.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { Suspense, SyntheticEvent, useEffect, useState } from 'react';
+import { Tab, Skeleton, Box } from '@mui/material';
+import { TabContext, TabList, TabPanel } from '@mui/lab';
+import { Await, useLoaderData } from 'react-router-dom';
+import { EngagementContent } from 'models/engagementContent';
+import { EngagementSummaryContent } from 'models/engagementSummaryContent';
+import { Editor } from 'react-draft-wysiwyg';
+import { getEditorStateFromRaw } from 'components/common/RichTextEditor/utils';
+import { combinePromises } from 'utils';
+import { Header2 } from 'components/common/Typography';
+import { colors } from 'components/common';
+
+export const EngagementContentTabs = () => {
+    const { content, contentSummary } = useLoaderData() as {
+        content: Promise<EngagementContent[]>;
+        contentSummary: Promise<EngagementSummaryContent[][]>;
+    };
+    const [selectedTab, setSelectedTab] = useState('0');
+    const handleChange = (event: SyntheticEvent<Element, Event>, newValue: string) => {
+        setSelectedTab(newValue);
+    };
+
+    const panelContents = combinePromises({ content, contentSummary });
+
+    const tabListRef = React.createRef<HTMLButtonElement>();
+
+    const checkFade = () => {
+        if (!tabListRef.current) return;
+        const scroller = tabListRef.current.getElementsByClassName('MuiTabs-scroller')[0];
+        const scrollPosition = scroller.scrollLeft; // distance from left edge
+        const maxScroll = scroller.scrollWidth - scroller.clientWidth; // distance from right edge
+        const fadeMargin = 64; // pixels
+        if (maxScroll - scrollPosition < fadeMargin) {
+            tabListRef?.current?.classList.remove('fade-right');
+        } else {
+            tabListRef?.current?.classList.add('fade-right');
+        }
+    };
+
+    const attachFade = () => {
+        if (tabListRef.current) {
+            const scroller = tabListRef.current.getElementsByClassName('MuiTabs-scroller')[0];
+            scroller.addEventListener('scroll', checkFade); // check when scrolling
+            const resizeObserver = new ResizeObserver(checkFade);
+            resizeObserver.observe(scroller); // check when window resizes
+            checkFade(); // initial check when attaching
+            return () => {
+                // cleanup
+                scroller.removeEventListener('scroll', checkFade);
+                resizeObserver.disconnect();
+            };
+        } else {
+            setTimeout(attachFade, 100); // try again in 100ms
+        }
+    };
+
+    useEffect(attachFade, [tabListRef]);
+
+    return (
+        <section id="content-tabs" aria-label="Engagement content tabs">
```

Comment:
Nice aria label for this region! We don't have a heading tag to lean on so I think this is adequately descriptive of the content to follow :)